DEHPCommon/UserPreferenceHandler/UserPreferenceService/UserPreferenceService.cs

```diff
@@ -0,0 +1,170 @@
+    using System.Reflection;
+
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Serialization;
+
+    using NLog;
+
+    /// <summary>
+    /// Definition of the <see cref="UserPreferenceService"/> used to load specific settings
+    /// </summary>
+    public class UserPreferenceService : IUserPreferenceService
+    {
+        /// <summary>
+        /// The logger for the current class
+        /// </summary>
+        private static Logger logger = LogManager.GetCurrentClassLogger();
+
+        /// <summary>
+        /// The path of the user preference directory storage
+        /// </summary>
+        public static string ApplicationExecutePath = Path.GetDirectoryName(Assembly.GetExecutingAssembly()?.Location);
+
+        /// <summary>
+        /// The name of folder where user preference is storage.
+        /// </summary>
+        public static string UserPreferenceDirectoryName = "UserPreferenceService";
+
+        /// <summary>
+        /// The setting file extension
+        /// </summary>
+        public const string SETTING_FILE_EXTENSION = ".settings.json";
+
+        /// <summary>
+        /// Gets or sets  user preference settings,
+        /// </summary>
+        public UserPreference UserPreferenceSettings { get; set; }
+
+        /// <summary>
+        /// Configuration user preference file path
+        /// </summary>
+        public string UserPreferenceDirectoryPath
+        {
+            get { return Path.Combine(ApplicationExecutePath, UserPreferenceDirectoryName); }
+        }
+        /// <summary>
+        /// Initializes a new instance of <see cref="UserPreferenceService"/>
+        /// </summary>
+        public UserPreferenceService()
+        {
+            this.UserPreferenceSettings = new UserPreference();
+        }
+
+        /// <summary>
+        /// Reads the <see cref="T"/> user preference in settings
+        /// </summary>
+        /// <typeparam name="T">A type of <see cref="UserPreference"/></typeparam>
+        /// <returns>
+        /// An instance of <see cref="UserPreference"/>
+        /// </returns>
+        public T Read<T>() where T : UserPreference
+        {
+            var assemblyName = this.QueryAssemblyTitle(typeof(T));
+
+            this.CheckConfigurationDirectory();
+
+            var path = Path.Combine(this.UserPreferenceDirectoryPath, assemblyName);
+
+            logger.Debug("Read user preference for {0} from {1}", assemblyName, path);
+
+            try
+            {
+                var fileExist = File.Exists($"{path}{SETTING_FILE_EXTENSION}");
+
+                if (!fileExist)
+                {
+                    this.Write(this.UserPreferenceSettings);
+                }
+
+                using (var file = File.OpenText($"{path}{SETTING_FILE_EXTENSION}"))
+                {
+                    var serializer = new JsonSerializer();
+                    var result = (List<ServerConnection>)serializer.Deserialize(file, typeof(List<ServerConnection>));
```

Comment:
Actually this will not work. You need to deserialize the whole file as one and just assign it to `this.UserPreferenceSettings`, not just server connections.